src/components/main.js

```diff
@@ -0,0 +1,39 @@
+import React, {useState} from "react";
+import {BrowserRouter as Router, Route, Switch} from "react-router-dom";
+import NavBar from './NavBar';
+import Home from './Home';
+import About from './About';
+import PyBuild from './Tools/pybuild';
+import CMDWord from './Tools/cmd-word';
+import DTEFileCreator from './Tools/dte-file-creator';
+import MemMap from './Tools/memmap';
+import MissShob from './Tools/miss-shob';
+
+const Main = () => {
+
+  const [minimizeSideBar, setMinimizeSideBar] = useState(true);
+  const MinMaxMain = () => {
```

Comment:
This name is not that clear...
I personally like to add `toggle` to functions that toggle stuff, something like `toggleMainWidth`, but it's up to you